pkg/controller/status/patch.go

```diff
@@ -0,0 +1,39 @@
+package status
+
+import (
+	"context"
+	"encoding/json"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type patchValue struct {
+	Op    string      `json:"op"`
+	Path  string      `json:"path"`
+	Value interface{} `json:"value"`
+}
+
+// PatchUpdateStatus performs the JSONPatch patch update to the Atlas Custom Resource.
+// The "jsonPatch" merge allows to update only status field so is more
+func PatchUpdateStatus(kubeClient client.Client, resource mdbv1.AtlasCustomResource) error {
+	return doPatch(kubeClient, resource, resource.GetStatus())
+}
+
+func doPatch(kubeClient client.Client, resource runtime.Object, statusValue interface{}) error {
+	payload := []patchValue{{
+		Op:    "replace",
+		Path:  "/status",
+		Value: statusValue,
+	}}
+
+	data, err := json.Marshal(payload)
+	if err != nil {
+		return err
+	}
+
+	patch := client.RawPatch(types.JSONPatchType, data)
+	return kubeClient.Status().Patch(context.Background(), resource, patch)
```

Comment:
If this context will be passed in the future, we should either expose it right away or use `context.TODO()`

Reply:
I thought about it and I wasn't sure if we need proper context support.
To my mind the context shines when there is a customer-facing system and we need to put time thresholds, responsiveness adjusting technics (so have possibility to cancel contexts and the whole pipe of requests). In this case the reconciliation is done in background fashion so it's ok if the request to Atlas has only HTTP timeouts and no extra orchestration.

Happy to discuss more

Reply:
I'm with anton in this one that if there's no need to cascade request and cancellation background is a good option

Reply:
My only concern here is hanging requests due to lack of timeouts. Not an issue if the client being used has reasonable default timeouts